ReleaseProcessScript/tools/Core/git_base_functions.ps1

```diff
@@ -267,30 +268,26 @@ function Get-Ancestor ($ExpectedAncestors)
   
   $Ancestors = git show-branch | Where-Object { $_.Contains('!') -eq $TRUE } | Where-Object { $_.Contains($Branchname) -ne $TRUE } | % {$_ -replace('.*\[(.*)\].*','$1')} | % { $_ -replace('[\^~].*','') }
 
-  $ReturnAncestor = $NULL
+  [System.Collections.ArrayList]$FoundAncestors = @()
   foreach ($Ancestor in $Ancestors) 
   {
     if ($ExpectedAncestors.Contains($Ancestor) -or $Ancestor.StartsWith($ExpectedAncestors))
     {
-      #In the List of Ancestors is more than one of the expected Ancestors. Commence to panic and ask the user
-      if ($ReturnAncestor -ne $NULL)
-      {
-        $ReturnAncestor = Read-Ancestor-Choice $ExpectedAncestors $ReturnAncestor
-        return $ReturnAncestor
-      }
-      else
-      {
-        $ReturnAncestor = $Ancestor
-      }
+      $FoundAncestors.Add($Ancestor) | Out-Null
     }
   }
 
-  if ($ReturnAncestor -eq $NULL)
+  if ($FoundAncestors.Count -eq 0)
+  {
+    return Read-Host "We expected some of the following Ancestors: '$ExpectedAncestors' but found none. Please enter the name of the Ancestor Branch"
+  }
+
+  if ($FoundAncestors.Count -gt 1)
```

Comment:
switch this and the last statement. returning the single item when -eq 1 would be more structed, going from 0 to 1 to many.